src/coreclr/src/System.Private.CoreLib/src/System/Array.CoreCLR.cs

```diff
@@ -150,8 +151,8 @@ public static unsafe void Copy(Array sourceArray, Array destinationArray, int le
                 (uint)length <= (nuint)destinationArray.LongLength)
             {
                 nuint byteCount = (uint)length * (nuint)pMT->ComponentSize;
-                ref byte src = ref sourceArray.GetRawSzArrayData();
-                ref byte dst = ref destinationArray.GetRawSzArrayData();
+                ref byte src = ref MemoryMarshal.GetRawArrayData(Unsafe.As<byte[]>(sourceArray));
```

Comment:
The array is not an actual `byte[]` array. It is a confusing to unsafe cast the array to `byte[]` when it is not actually `byte[]`

Reply:
Add a comment at least.

Reply:
Would you prefer something like `Unsafe.As<T, byte>(ref GetRawArrayData(...))` instead?

Reply:
> `Unsafe.As<T, byte>(ref GetRawArrayData(...))`

I am not sure how it would here. There is no `T` available here.

The fact that layout is same for all SZ arrays was used as an implementation detail of GetRawArrayData before this change. Now, it is leaking out silently here.

What about changing this to `ref byte src = ref Unsafe.As<RawArrayData>(sourceArray);`? It would make me happy because it makes it obvious that it is tightly coupled with array layout.